### PR TITLE
Ambush choke in combat

### PR DESCRIPTION
**Pull request by charleyc-dr**
(Plus a script for filling a dirt stacker)

### CODE REVIEW

**Comment by Sarvatt** [comment]
Could you add this to https://github.com/rpherbig/dr-scripts/blob/master/validate.lic#L344 ?

File: combat-trainer.lic [hunk 1]
```diff
@@ -1876,6 +1881,40 @@ class TrainerProcess
...
+      ) != 'Roundtime'
+      pause
+      waitrt?
+    end
+    if dirt_in_hand
+      bput("put my dirt in my #{@dirt_stacker}",
+           'dumping some dirt', 'What were you referring')
+    end
+
+    false
```
Comment (rpherbig) [inline combat-trainer.lic]: I realize you're following the precedent of the other training abilities, but I'm not convinced the return value is used anywhere. What do you think?

File: fill-dirt.lic [hunk 2]
```diff
@@ -0,0 +1,31 @@
+custom_require.call(%w[common common-items drinfomon])
```
Comment (rpherbig) [inline fill-dirt.lic]: Do you plan to document this on elanthipedia?

File: fill-dirt.lic [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+                             'over three quarters full',
+                             'over half full',
+                             'over a quarter full',
+                             'has a little dirt',
+                             'out of dirt') == 'it is full of dirt'
+  end
+
+  def fill_stacker
+    loop do
+      forage?('dirt')
```
Comment (rpherbig) [inline fill-dirt.lic]: Do you want to break if `forage?` fails?

File: combat-trainer.lic (hunk 1 above)
Comment (rcuhljr) [reply combat-trainer.lic]: You'd need a return up in the case that calls this like `ambush_stun` has, but that's only necessary if the action here could cause a real change in game state, I don't think choke can kill mobs so there's no need.

File: fill-dirt.lic (hunk 2 above)
Comment (charleyc-dr) [reply fill-dirt.lic]: Sure, I can do that!

File: combat-trainer.lic (hunk 1 above)
Comment (charleyc-dr) [reply combat-trainer.lic]: You're right, totally not needed.  Removed.

File: fill-dirt.lic (hunk 3 above)
Comment (charleyc-dr) [reply fill-dirt.lic]: Good idea; done.

File: fill-dirt.lic (hunk 2 above)
Comment (rpherbig) [reply fill-dirt.lic]: Can you add a header documentation block to the script?

Comment (charleyc-dr) [reply fill-dirt.lic]: Done!

**Pull request merged**